.changeset/silent-carrots-greet.md

```diff
@@ -0,0 +1,12 @@
+---
+"@salt-ds/lab": minor
+---
+
+Updated the `NumberInput`:
+
+1. Added support for `format` and `parse` callbacks for formatting capabilities.
+2. Added `clampValue` prop to allow enforcement of values within the min and max range.
+3. Refactored `stepBlock` prop to be `stepMultiplier` to be consistent with `Slider` and `RangeSlider` components and to ensure that valid values remain reachable.
+4. Refactored `decimalPlaces` to be `decimalScale`.
```

Comment:
see the changelog for carousel
you can diff with a code example
https://github.com/jpmorganchase/salt-ds/blob/baebcea6808662d45c65641cf2fb8b5db2b69f7d/.changeset/dirty-months-sip.md
Might make the changes easier to consume

Whilst labs are un-stable components we can mitigate the friction by providing great changelogs